pkg/vpwned/server/vjailbreak_proxy.go

```diff
@@ -496,3 +499,170 @@ func (p *vjailbreakProxy) RevalidateCredentials(ctx context.Context, in *api.Rev
 		return nil, fmt.Errorf("unknown credentials kind: %s", kind)
 	}
 }
+
+func (p *vjailbreakProxy) InjectEnvVariables(ctx context.Context, in *api.InjectEnvVariablesRequest) (*api.InjectEnvVariablesResponse, error) {
+	const fn = "InjectEnvVariables"
+	logrus.WithFields(logrus.Fields{
+		"func":        fn,
+		"http_proxy":  in.GetHttpProxy(),
+		"https_proxy": in.GetHttpsProxy(),
+		"no_proxy":    in.GetNoProxy(),
+	}).Info("Starting InjectEnvVariables request")
+	defer logrus.WithField("func", fn).Info("Completed InjectEnvVariables request")
+
+	const (
+		configMapName      = "pf9-env"
+		configMapNamespace = "migration-system"
+		deploymentName     = "migration-controller-manager"
```

Comment:
you would also want to restart other pods where pf9-env is mounted. does proxy pod mounts pf9-env?